test/test_limits.py

```diff
@@ -500,6 +500,25 @@ async def test_reserve_pool_size(pg, bouncer):
         await result
 
 
+@pytest.mark.asyncio
+async def test_user_reserve_pool_size(pg, bouncer):
+    bouncer.admin("set reserve_pool_timeout = 2")
+
+    # Disable tls to get more consistent timings
+    bouncer.admin("set server_tls_sslmode = disable")
+
+    with bouncer.log_contains("taking connection from reserve_pool", times=2):
+        # respoolsize1 user has a pool_size of 1 and reserve_pool_size of 2
+        # this means 1 connection should happen immediately while 2 out of
+        # the 3 remaining connections happen after reserve_pool_timeout
+        result = bouncer.asleep(10, dbname="p0a", user="respoolsize1", times=4)
```

Comment:
Why is this one using `p0a` and the other ones `p0`?

Reply:
p0 has the bouncer user configured to it while p0a does not have any user. From my testing it seems like no matter who I am connecting as to p0, it will apply the pool settings of the configured user.